frontend/src/components/normal-playlist/playlist.js

```diff
@@ -0,0 +1,257 @@
+                    selectedArtists: this.state.selectedArtists.concat(data.artists.items[0].name)
+                })
+            }
+        },
+
+        (err) => {
+            console.log(err)
+        }
+    );
+  }
+
+  //this is for creating playlist
+  searchArtist = (artist) => {
+    artist.map((artistName, key) => {
+        spotifyApi.searchArtists(artistName).then(
+            (data) => {
+                if(data !== undefined) {
+                    this.setState({
+                        artist:{name: data.artists.items[0].name, id : data.artists.items[0].id}  // index is 0 because most popular query's index is 0 
+                    }, () => {
+                        this.getTopTracks(this.state.artist.id, 'TR'); // a callback, will be executed after setting state
+                    });
+                }          
+            },
+            (err) => {
+                console.log(err)
+            }
+        );
+    })
+  }
+
+
+  //tracks that we are going to add to playlist
+  getTopTracks = (artistId, country) => {
+      spotifyApi.getArtistTopTracks(artistId, country).then(     
+        (data) => {
+            data.tracks.map((track) => {
+                this.setState({
+                    topTracksIds:this.state.topTracksIds.concat(track.uri) //storing track id's in an array
+                })
+            })
+
+            this.addTracksToPlaylist(this.state.user.playlistId, this.state.topTracksIds)
+          },
+
+        (err) => {
+            console.log(err)
+        }
+      )
+  }
+
+  addTracksToPlaylist = (playlistId, topTracksIds) => {
+    spotifyApi.addTracksToPlaylist(playlistId, topTracksIds).then(
+        (data) => {
+          //this.resetIframe()
+        },
+        (err) => {
+            console.log(err)
+        }
+    )
+
+    this.setState({
+        topTracksIds: []
+    })
+  }
+
+  handleSelect = (value) => {
+    this.setState({
+        selectedArtists: this.state.selectedArtists.concat(value)
+    }, () => {
+        console.log(this.state.selectedArtists)
+    }) 
+  }
+
+  handleDeselect = (value) => {
+    this.setState({
+        selectedArtists: this.state.selectedArtists.filter(name => !name.includes(value))
+    }, () => {
+        console.log(this.state.selectedArtists)
+    });
+
+
+  }
+
+  handleCreate = () => {
+    this.createPlaylist(this.state.user.id, this.state.playlistName, this.state.playlistDescription);
+    this.setState({
+        displayPlaylist: 'block',
+      })
+
+    setTimeout(() => { // iframe needs a reload to show songs
+        this.setState({ random: this.state.random + 1 })
+    }, 700);
+
+    setTimeout(() => { // selected songs dissapear 
+        this.setState({
+            selectedArtists: [],
+            topTracksIds: []
+        })
+    },3000)
+  }
+
+  createPlaylist = (userId, name, description) => {
+    if(name === undefined || description === undefined) {
```

Comment:
This still isnt working as expected as name is just an empty string meaning that it is defined. Perhaps try checking to see if the string is not only undefined but empty.

Reply:
```
    if(!name.length > 0) {
        name = 'Playlist App';
        description= 'Created with Playlist App!'
    }
    if(!description.length > 0) {
        description = 'Created with Playlist App'
    }
 
```

Reply:
I couldn't recreate the bug 'cause of that  I don't know it's working or not.

Reply:
It looks like you fixed it but I was able to find another one. If the user enters the same artist twice, it shows up twice when selecting the artist. [Here](https://imgur.com/a/bx52Ghf) is an example.